Neos.Neos/Classes/FrontendRouting/DimensionResolution/DelegatingResolver.php

```diff
@@ -66,13 +59,9 @@ public function fromRequestToDimensionSpacePoint(
     public function fromDimensionSpacePointToUriConstraints(
         DimensionSpacePoint $filteredDimensionSpacePoint,
         DocumentNodeInfo $targetNodeInfo,
-        UriConstraints $uriConstraints
+        Site $targetSite,
+        UriConstraints $uriConstraints,
     ): UriConstraints {
-        $targetSite = $this->siteRepository->findOneByNodeName($targetNodeInfo->getSiteNodeName());
-
-        if ($targetSite === null) {
-            throw new \RuntimeException('Did not find site object for identifier ' . $targetNodeInfo->getSiteNodeName()->value);
-        }
         $targetSiteConfiguration = $targetSite->getConfiguration();
```

Comment:
@bwaidelich I was thinking if it would be better to just pass the SiteConfiguration here with the interface. This would make it easier to test. WDYT?

Reply:
Wasn't there a resolver that depends on the site's _domain_? For that we need the Site instance AFAIS

Reply:
No, that was the [CrossSiteLinker](https://github.com/neos/neos-development-collection/blob/4e6b69370f119c3dae45d12f04426ab3fb27f3af/Neos.Neos/Classes/FrontendRouting/CrossSiteLinking/CrossSiteLinker.php#L46)

Reply:
Ah right.. But _could_ there be a scenario that requires the site itself?

Reply:
I was wondering the same. IDK... But then "premature optimization"?